README.md
# Microninja
## Microninja-settings

Micronja-settings fornisce tutti gli strumenti necessari per la configurazione ed il corretto funzionamento del sistema. Con questo pacchetto puoi collegare il sistema ad internet, configurare mouse, tastiera, wallpaper, schermo, e impostare il livello di sicurezza della navigazione tramite il parental control.

### Tastiera

Permette di configurare la tastiera indicando la nazionalità e relative variazioni. Le impostazioni predefinite sono quelle ottimali per la tastiera Microninja.

### Mouse

Configurazione della velocità del mouse: bassa, media, alta.

### Audio

L'uscita audio può essere selezionata fra quella analogica (altoparlanti o auricolari) o quella HDMI (tv o monitor).

### Schermo

Tramite queste impostazioni è possibile scegliere la risoluzione dello schermo. L'elenco delle modalità disponibili dipende dal monitor collegato al tuo Microninja.

### WiFi

Strumento di configurazione della connessione internet. Permette di selezionare fra: ethernet, wifi o condivisione della connessione mobile via USB.

### Ripristino
Ripristina le configurazioni iniziali, riportando il tuo Microninja alle impostazioni di fabbrica.

### Grafica

Permette di selezionare lo sfondo del desktop, di scegliere il salvaschermo e di ripristinare le impostazioni predefinite.

### Carattere
Permette la selezione delle dimensioni predefinite dei caratteri (Piccole, medie, grandi)

### Parental control
Permette di abilitare e configurare le impostazioni di controllo parentale dei contenuti visualizzabili dal tuo Microninja

### Informazioni
Visualizza informazioni sul tuo Microninja, fra cui il modello, lo spazio disco e la versione software.


debian/changelog
microninja-settings (2.6.6) unstable; urgency=low

  * Dependencies bugfix

 -- Alfabook srl <info@scuolabook.it>  Mon, 11 Dec 2017 12:58:00 +0000

microninja-settings (2.6.5) unstable; urgency=low

  * Added microninja-locale dependency

 -- Alfabook srl <info@scuolabook.it>  Mon, 11 Dec 2017 12:58:00 +0000

microninja-settings (2.6.4) unstable; urgency=low

  * Added microninja-toolset dependency

 -- Alfabook srl <info@scuolabook.it>  Mon, 11 Dec 2017 12:58:00 +0000

microninja-settings (2.6.3) unstable; urgency=low

  * Bugfix in plugin localization (added locale setting in show menu)

 -- Alfabook srl <info@scuolabook.it>  Mon, 11 Dec 2017 12:58:00 +0000

microninja-settings (2.6.2) unstable; urgency=low

  * Addng translation for account.py reboot message

 -- Scuolabook srl <info@scuolabook.it>  Mon, 11 Dec 2017 12:58:00 +0000

microninja-settings (2.6.1) unstable; urgency=low

  * Status bar tooltip internet plugin

 -- Scuolabook srl <info@scuolabook.it>  Mon, 4 Dec 2017 12:58:00 +0000

microninja-settings (2.6.0) unstable; urgency=low

  * Removing menu cache for localization

 -- Scuolabook srl <info@scuolabook.it>  Mon, 4 Dec 2017 12:58:00 +0000

microninja-settings (2.5.9) unstable; urgency=low

  * Adding Portuguese, internationalization ready

 -- Scuolabook srl <info@scuolabook.it>  Thu, 17 Nov 2017 12:58:00 +0000

microninja-settings (2.5.8) unstable; urgency=low

  * Preparing for internationalization / localization

 -- Scuolabook srl <info@scuolabook.it>  Mon, 14 Nov 2017 12:58:00 +0000

microninja-settings (2.5.7) unstable; urgency=low

  * Changed information icon

 -- Scuolabook srl <info@scuolabook.it>  Mon, 9 Oct 2017 12:58:00 +0000

microninja-settings (2.5.6) unstable; urgency=low

  * Translation fix

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.5.5) unstable; urgency=low

  * Fixed mouse blocked during restart request

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.5.4) unstable; urgency=low

  * Button re aligned in users

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.5.3) unstable; urgency=low

  * Calling microninja.script

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.5.2) unstable; urgency=low

  * Pi3 recognized; spelling mistake on resetting system

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.5.1) unstable; urgency=low

  * Added reset to factory button

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.5) unstable; urgency=low

  * Italian translation for counties in keyboard configuration

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.4) unstable; urgency=low

  * Profile bugfixes

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.3) unstable; urgency=low

  * New profile

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

microninja-settings (2.2-2) unstable; urgency=low

  * Changes on Parental Control

 -- Scuolabook srl <info@scuolabook.it>  Mon, 5 Apr 2016 12:58:00 +0000

kano-settings (2.2-1) unstable; urgency=low

  * Adding Gtk.Plug support

 -- Team Kano <dev@kano.me>  Wed, 26 Aug 2015 17:22:00 +0000

kano-settings (2.0-3) unstable; urgency=low

  * Add Midori/Youtube cookie functionality

 -- Team Kano <dev@kano.me>  Mon, 29 Jun 2015 14:49:00 +0000

kano-settings (2.0-2) unstable; urgency=low

  * Add ability to set hostname

 -- Team Kano <dev@kano.me>  Fri, 24 Apr 2015 15:49:00 +0000

kano-settings (2.0-1) unstable; urgency=low

  * Adding the new background selector for the initflow

 -- Team Kano <dev@kano.me>  Fri, 24 Apr 2015 15:49:00 +0000

kano-settings (1.3-3) unstable; urgency=low

  * support for the new kano-init

 -- Team Kano <dev@kano.me>  Fri, 27 Mar 2015 15:05:00 +0000

kano-settings (1.3-2) unstable; urgency=low

  * Added Option to restore users customized LXPanels to original settings

 -- Team Kano <dev@kano.me>  Wed, 26 Nov 2014 14:45:00 +0000

kano-settings (1.3-2) unstable; urgency=low

  * Allow kano-settings to close splash screens.

 -- Team Kano <dev@kano.me>  Wed, 26 Nov 2014 14:45:00 +0000

kano-settings (1.3-1) unstable; urgency=low

  * Added No Internet screen

 -- Team Kano <dev@kano.me>  Wed, 26 Nov 2014 14:45:00 +0000

kano-settings (1.2.2-1) unstable; urgency=low

  * Refactoring system-related functions

 -- Team Kano <dev@kano.me>  Wed, 18 Sep 2014 11:45:00 +0000

kano-settings (1.1-2) unstable; urgency=low

  * kano-uixinit removed

 -- Team Kano <dev@kano.me>  Wed, 13 May 2014 18:11:00 +0000

kano-settings (1.1-1) unstable; urgency=low

  * Merged with the kano-connect scripts
  * The connect scripts are now a subpackage of kano-settings

 -- Team Kano <dev@kano.me>  Wed, 26 Mar 2014 15:59:19 +0000

kano-settings (1.0-7) unstable; urgency=low

  * Package rebuilt, updated to revision 77f4b499.

 -- Team Kano <dev@kano.me>  Tue, 04 Mar 2014 11:00:32 +0000

kano-settings (1.0-6) unstable; urgency=low

  * Package rebuilt, updated to revision bb1c6e46.

 -- Team Kano <dev@kano.me>  Mon, 03 Mar 2014 21:02:12 +0000

kano-settings (1.0-5) unstable; urgency=low

  * Package rebuilt, updated to revision 54f93581.

 -- Team Kano <dev@kano.me>  Mon, 03 Mar 2014 17:31:06 +0000

kano-settings (1.0-4) unstable; urgency=low

  * Package rebuilt, updated to revision 789f27ea.

 -- Team Kano <dev@kano.me>  Mon, 03 Mar 2014 16:48:55 +0000

kano-settings (1.0-3) unstable; urgency=low

  * Package rebuilt, updated to revision 971a910c.

 -- Team Kano <dev@kano.me>  Mon, 03 Mar 2014 14:18:10 +0000

kano-settings (1.0-2) unstable; urgency=low

  [ Team Kano ]
  * Package rebuilt, updated to revision 52a5cb59.

 -- Team Kano <dev@kano.me>  Mon, 03 Mar 2014 13:59:12 +0000


microninja_settings/screens.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Copyright (C) 2015 Kano Computing Ltd.
# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
#
# Structures to hold the screens
#
# Copyright (C) 2016 Alfabook srl
# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
# rebadged with microninja
# Italian translation
# notifications removed
# overclock removed

from collections import OrderedDict

from microninja.utils import get_user_unsudoed

import microninja_settings.common as common
from microninja_settings.config_file import get_setting
from microninja_settings.components.menu_button import Menu_button

import microninja_settings.system.keyboard_layouts as keyboard_layouts
from microninja_settings.set_keyboard import choose_keyboard_screen, keyboard_screen_install
from microninja_settings.set_mouse import SetMouse
#from microninja_settings.set_notifications import SetNotifications
from microninja_settings.set_font import SetFont
from microninja_settings.set_audio import SetAudio
from microninja_settings.set_display import SetDisplay
from microninja_settings.set_wifi import SetWifi, SetProxy
from microninja_settings.no_internet_screen import NoInternet
from microninja_settings.set_overclock import SetOverclock
from microninja_settings.set_account import SetAccount
from microninja_settings.set_about import SetAbout
from microninja_settings.set_advanced import SetAdvanced, SetPassword
from microninja_settings.set_style import SetStyle
from microninja_settings.set_wallpaper import FirstBootSetWallpaper
from microninja_settings.locale import LocaleConfig


class Screen(object):
    """
    NB: screen_no is here for legacy reasons - remove as soon as support is
        withdrawn.
    """

    def __init__(self, name, label, screen_widget,
                 screen_no=None, on_home_screen=True,
                 setting_param=None, icon=''):
        self.name = name
        self._label = label
        self.screen_widget = screen_widget
        self.screen_no = screen_no
        self.on_home_screen = on_home_screen
        self.menu_button = None
        self._setting_param = setting_param
	self._icon = icon
	if(self._icon == ''):
		self._icon = label

    def create_menu_button(self, cb):
        self.menu_button = button = Menu_button(self._label, '', icon=self._icon)
        button.button.state = self.screen_no
        button.button.connect('clicked', cb, self.name)

    def refresh_menu_button(self):
        description = ''

        if self.name == 'appearance':
             description = _('Wallpapers and screensavers')
             self.menu_button.description.set_text(description)
             return

        if self._setting_param:
            if self.name == 'font':
                value = get_setting(self._setting_param)
                if value == 0:
                   description = _('Small')
                if value == 1:
                   description = _('Medium')
                if value == 2:
                   description = _('Big')
            elif self.name == 'keyboard':
                continent = get_setting('Keyboard-continent-index')
                country = get_setting('Keyboard-country-index')
                description = keyboard_layouts.continents_list_countries[continent][country]
            else:    
                description = get_setting(self._setting_param)
        else:
            if self.name == 'wifi':
                if common.has_internet:
                    description = _('Connected')
                else:
                    description = _('Not connected')

            elif self.name == 'account':
                description = get_user_unsudoed()
            
            elif self.name == 'display':
                return

        self.menu_button.description.set_text(description)


class ScreenCollection(OrderedDict):

    def __init__(self, screens):
        super(ScreenCollection, self).__init__()

        for screen in screens:
            self[screen.name] = screen

    def get_screen_from_number(self, number):
        for screen in self.itervalues():
            if screen.screen_no == number:
                return screen

    def get_screens_on_home(self):
        displayed_screens = []

        for screen in self.itervalues():
            if screen.on_home_screen:
                displayed_screens.append(screen)

        return displayed_screens


SCREENS = ScreenCollection([
    Screen('keyboard', _('Keyboard'), choose_keyboard_screen, screen_no=0,
           setting_param='Keyboard-country-index', icon='Keyboard'),
    #Screen('mouse', 'Mouse', SetMouse, screen_no=1, setting_param='Mouse', icon='Mouse')
    Screen('locale', _('Language'), LocaleConfig, screen_no=1, setting_param='Locale', icon='Language'),
    Screen('audio', _('Audio'), SetAudio, screen_no=2, setting_param='Audio', icon='Audio'),
    Screen('display', _('Screen'), SetDisplay, screen_no=3, icon='Display'),
    Screen('wifi', _('WiFi'), SetWifi, screen_no=4, icon='WiFi'),
    #Screen('overclocking', 'Overclocking', SetOverclock, screen_no=5,
    #       setting_param='Overclocking', icon='Overclocking'),
    Screen('account', _('Reset'), SetAccount, screen_no=5, icon='Account'),
    Screen('appearance', _('Style'), SetStyle, screen_no=6,
           setting_param='Wallpaper', icon='Style'),
    Screen('font', _('Fonts'), SetFont, screen_no=7, setting_param='Font', icon='Font'),
    Screen('advanced', _('Parental Control'), SetAdvanced, screen_no=8, icon='Advanced'),
    Screen('about', _('About'), SetAbout, screen_no=9, icon='About'),
    #Screen('notifications', 'Notifications', SetNotifications, screen_no=11),
    Screen('no-internet', 'No-internet', NoInternet, screen_no=10,
           on_home_screen=False),
    Screen('proxy', 'proxy', SetProxy, screen_no=11, on_home_screen=False),
    Screen('first-boot-set-wallpaper', 'first-boot-set-wallpaper',
           FirstBootSetWallpaper, screen_no=12, on_home_screen=False),
    Screen('set-parental-password', 'set-parental-password', SetPassword,
           screen_no=13, on_home_screen=False),
    #special keyboard window for first boot
    Screen('keyboard-install', _('Keyboard'), keyboard_screen_install, screen_no=14,
           setting_param='Keyboard-country-human', icon='Keyboard', on_home_screen=False),
    # TODO: Add 'Locale' screen to home screen when translations are available.
    # TODO: Add icon for the 'Locale' screen.
])
